test/test_main.py

```diff
@@ -0,0 +1,199 @@
+valid_secret_pass_request = {"name": "projects/project_id/secrets/secret_password/versions/latest"}
+def fake_secret_client(request):
+    if request == valid_secret_user_request:
+        username=MagicMock()
+        username.payload.data = b'username'
+        return username
+
+    elif request == valid_secret_pass_request:
+        password=MagicMock()
+        password.payload.data = b'password'
+        return password
+
+    else:
+        raise RuntimeError('secret does not exist!')
+        
+
+#--------------------------------------------------------------------------------unit test for get_wzdx_schema function--------------------------------------------------------------------------------
+def test_get_wzdx_schema():
+    expected_schema= json.loads(open('translator/sample files/validation_schema/wzdx_v3.0_feed.json').read())
+    actual=main.get_wzdx_schema('translator/sample files/validation_schema/wzdx_v3.0_feed.json')
+    assert actual == expected_schema
+
+
+def test_get_wzdx_schema_invalid_data():
+
+    with pytest.raises(RuntimeError) as runtimeErr:
+        main.get_wzdx_schema('test/docs/invalid_schema.json')
+    assert 'invalid schema: not valid json' in str(runtimeErr.value)
+
+def test_get_wzdx_schema_not_exist():
+        
+    with pytest.raises(RuntimeError) as runtimeErr:
+        main.get_wzdx_schema('not_exist.json')
+    assert 'invalid schema: file does not exist' in str(runtimeErr.value)   
+    
+
+#--------------------------------------------------------------------------------unit test for unsupported_messages_callback function--------------------------------------------------------------------------------
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.dict(os.environ, {'project_id': 'project_id',
+    'unsupported_messages_topic_id': 'unsupported_messages_topic_id'})
+def test_unsupported_messages_callback_dict(pubsub):
+
+    output=main.unsupported_messages_callback({'messages':'unsupported_messages'})
+    publisher=pubsub().publish
+    publisher.assert_called_with(pubsub().topic_path('project_id', 'unsupported_messages_topic_id'),str.encode(json.dumps({'messages':'unsupported_messages'}, indent=2)),origin='auto_icone_translator_ftp cloud function')
+
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.dict(os.environ, {'project_id': 'project_id',
+    'unsupported_messages_topic_id': 'unsupported_messages_topic_id'})
+def test_unsupported_messages_callback_string(pubsub):
+
+    output=main.unsupported_messages_callback('unsupported_messages')
+    publisher=pubsub().publish
+    publisher.assert_called_with(pubsub().topic_path('project_id', 'unsupported_messages_topic_id'),str.encode(json.dumps('unsupported_messages')),origin='auto_icone_translator_ftp cloud function')
+
+
+@patch('google.cloud.pubsub_v1.PublisherClient')
+@patch.dict(os.environ, {})
+def test_unsupported_messages_callback_invalid(pubsub):
```

Comment:
This method is really testing the lack of environment variable. Can we rename it such?

Reply:
i renamed it and pushed it to dev